scrape/target.go

```diff
@@ -273,6 +283,31 @@ func (t *Target) Health() TargetHealth {
 	return t.health
 }
 
+// IntervalAndTimeout returns the interval and timeout derived from
+// the targets labels.
+func (t *Target) IntervalAndTimeout(prevInterval, prevDuration time.Duration) (time.Duration, time.Duration, error) {
+	t.mtx.RLock()
+	defer t.mtx.RUnlock()
+
+	intervalLabel := t.labels.Get(model.ScrapeIntervalLabel)
+	interval, err := model.ParseDuration(intervalLabel)
+	if err != nil {
+		return prevInterval, prevDuration, errors.Errorf("Error parsing interval label %q: %v", intervalLabel, err)
+	}
+	timeoutLabel := t.labels.Get(model.ScrapeTimeoutLabel)
+	timeout, err := model.ParseDuration(timeoutLabel)
+	if err != nil {
+		return prevInterval, prevDuration, errors.Errorf("Error parsing timeout label %q: %v", timeoutLabel, err)
+	}
+
+	return time.Duration(interval), time.Duration(timeout), nil
+}
```

Comment:
I still don't exactly like how this function takes the previous interval and duration as arguments. The idea was to skip an error check (in case of a cosmic ray) by returning the old values on error, and it is handy:

https://github.com/prometheus/prometheus/blob/315031aa6f376606567375d511bf67982a9535ac/scrape/scrape.go#L523-L536

but at the cost of things being weird if this function is used in any other scenario (which it isn't currently).

Reply:
let's unexport this function

Reply:
Instead of pref, should we call this default?

Reply:
friendly ping :)

Reply:
Sorry, I was waiting on a response to https://github.com/prometheus/prometheus/pull/8911#discussion_r673055312